mne/source_estimate.py

```diff
@@ -3781,7 +3781,7 @@ def stc_near_sensors(
     subjects_dir=None,
     src=None,
     picks=None,
-    surface="pial",
+    surface=None,
```

Comment:
IIRC there was logic to the original choice -- source spaces are almost always created with white as the surface (for inverse modeling) but non-inverse-based projections like this should almost always use pial. So there was some logic to it. Not sure we should make this change

Reply:
Why would you model with white and then project to pial?

Reply:
That seems very confusing. Maybe you shouldn't be allowed to pass a surface at all but should just use the surface in the source space?

Reply:
Hmm I understand but this odd choice is causing problems hence raising the issue.

Reply:
Did you read/see my suggestion above? You haven't responded to it:

> If we do change it, I think we need to properly deprecate because people like me would otherwise silently get a different (wrong!) result, namely projection to the white surface whereas before it projected (correctly) to pial when I ask for a decimated version by passing src.

This would be one way I think for you to get the behavior/consistency you want without silently breaking existing use cases that give correct outputs on `main` but would give incorrect outputs on this PR.

Reply:
Sure, I can do that, sorry in discussing why it was that way, I lost the thread that you wouldn't mind changing it with a deprecation cycle.

Reply:
How should this be deprecated? Default ``surface=None`` but if  ``surface='pial'`` do a deprecation warning? Or just the deprecation warning on ``'pial'`` but don't change the default until a full cycle?

Reply:
For deprecations the policy is to preserve the old/current behavior for the upcoming release (1.7), but warn that the default will change in the next *only if the behavior will actually change* in the next release (1.8) given the current function inputs being passed.

So for example you would warn if someone did `setup_source_space(subject)` (which defaults to `surface='white'`) and `stc_near_sensors(evoked, src=src)` (because in 1.7 the default is `pial` but in 1.8 the behavior under `None` would change to `white`); but you would not warn if they did `setup_source_space(subject, surface='pial')` then `stc_near_sensors(evoked, src=src)` because the behavior would be the same in 1.7 and 1.8. I think this can be accomplished with pseudocode:
```
def stc_near_sensors(..., surf=None):
    ...
    if surf is None:
        if src is not None:
            src_surf = ...
            if src_surf != "pial"
                warn(..., DeprecationWarning)
        surf = "pial"
```
then after 1.7 comes out we simplify to
```
    if surf is None:
        if src is None:
            surf = "pial"
        else:
            surf = ... # src_surf
```

Reply:
I'm not sure that the surface is stored in the `src` instance. I looked and couldn't find it and it looks from `setup_source_space` that the surface is not stored as far as I can tell... so then you can't fill in the `src_surf = ....` line. Any ideas on how to handle that?

Reply:
You could `try/except` to load the `lh.pial` surface. If it raises an exception (like file not found) or `not np.allclose(src[0]['rr'], pial_rr)` you could conclude it's not `pial`. Hopefully not too painful (5ish lines?) but would do the trick I think.

Reply:
Yikes okay well at least it will only in the codebase for a version

Reply:
Ok, I didn't end up doing it exactly how you suggested but I did it so that if you had no argument for surface and passed a src then you would get a warning that the behavior will change which is what you asked for.